modules/linear_boltzmann_solvers/a_lbs_solver/lbs_solver.cc

```diff
@@ -644,6 +656,19 @@ LBSSolver::SetOptions(const InputParameters& params)
         SetBoundaryOptions(bndry_params);
       }
     }
+
+    else if (spec.Name() == "point_sources")
+    {
+      spec.RequireBlockTypeIs(ParameterBlockType::ARRAY);
+      for (const auto& sub_param : spec)
+      {
+        point_sources_.push_back(Chi::GetStackItem<PointSource>(
+          Chi::object_stack, sub_param.GetValue<size_t>(), __FUNCTION__));
+
+        // If the discretization is defined, initialization is possible
+        if (discretization_) point_sources_.back().Initialize(*this);
```

Comment:
It is more clear when you look at the `PointSource::Initialize` routine. When we initialize a point source, we go through each cell, find which cells the point source belongs to, and then map the point source to the finite element space via weight based on where the point source is in a cell. Since the solver creates its own discretization during its `Initialize` routine, we cannot initialize a point source until a discretization exists. This is placed here so that if a point source is added at some time after the main `Initialize` routine is called, the point source can be initialized without having to reinitialize everything. Most of the `Initialize<something>` routines are protected and not callable externally.